### DISCUSSION TITLE
App Crashes after opening screen with map for the second time

### DISCUSSION THREAD

**Discussion by Arilith** [rnmapbox/maps#2101] [Q&A]
Hi there,

Currently, I'm rewriting my web-application to a React Native application using Expo. I've managed to get the map working and rendering correctly when I open the correct screen.

![image](https://user-images.githubusercontent.com/13538135/183287569-19791193-4991-422d-9026-259cd41ad033.png)

However, when I navigate away from this screen (using tab navigation) and then open it back up again, the app just freezes and later Android tells me that the app is not responding.

I first thought it was because I had `detachInactiveScreens={false}` on my Stack Navigator, but disabling that doesn't change the problem. 

I'm using the MapBox implementation of the map., if that makes a difference at all.

Could this be due to emulation? I don't have an physical android device to test on right now. I've looked on Stack Overflow and found something about having components in my map, however, even without any components in my map, it still does the same thing.

My current code:
```react

import React from 'react'
import { Text, View, StyleSheet } from 'react-native'
import { GlobalStyles } from '../../../Constants/theme';

import MapboxGL from '@rnmapbox/maps';

MapboxGL.setAccessToken('pk.TOKEN');
export const MapScreen = () => {
  return (
    <View style={ styles.container }>
      <MapboxGL.MapView styleURL='mapbox://styles/arilith/ckr9r509238t817mxz75lwbxt' style={ styles.map }>


      </MapboxGL.MapView>
    </View>
  )
}

const styles = StyleSheet.create({
  container: {
    ...GlobalStyles.pageContainer,
    padding: 0,
  },
  innerContainer: {
    flex: 1,
    justifyContent: 'center',
    padding: 40,
  },
  map: {
    flex: 1,
  }
});
```

Thanks a lot in advance!

**Comment by Arilith** [comment]
After some further investigation, this problem does **not** occur when the map is as a screen in a stack view while switching back and forwards. It looks like it has something to do with the use of the Tab Navigator...

EDIT: I've been able to solve the problem by doing the following:
![image](https://user-images.githubusercontent.com/13538135/183293200-9afdb228-29be-4b09-a6df-af5d97970966.png)
![image](https://user-images.githubusercontent.com/13538135/183293204-f4c1091c-8a3b-4ebe-a507-b1ba1f053e8d.png)
But this seems kind of a hacky/work-around way of doing things.

**Reply by felixgourdeau** [reply]
See https://github.com/rnmapbox/maps/issues/1958 for more information
